lib/qio_field.cpp

```diff
@@ -488,8 +428,7 @@ void write_spinor_field(const char *filename, void *V[], QudaPrecision precision
 
   /* Read the spinor field record */
   printfQuda("%s: writing %d vector fields\n", __func__, Nvec); fflush(stdout);
-  int status
-    = write_field(outfile, 2 * nSpin * nColor, Nvec, V, precision, precision, subset, parity, nSpin, nColor, type);
+  int status = write_field(outfile, Nvec, V, precision, precision, subset, parity, nSpin, nColor, 2 * nSpin * nColor, type);
```

Comment:
What was the reason for this change?

Reply:
Previously, the QIO interface used template functions `vput<vlen>` and `vget<vlen>`, helper functions that are passed to QIO for packing and unpacking.  This limited the range of possible `nColor` values we could use QIO with to whatever the dev had pre-instantiated.  I realised a better approach was make these functions purely runtime, utilizing a static runtime value for `vlen` instead of templating it.  This allowed for a signficant cleanup here.

Reply:
Out of scope for this release. These are historical, and have been used for verification only. These should be cleaned up at some point.